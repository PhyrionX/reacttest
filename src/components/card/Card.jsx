import React from 'react';
import './card.scss';

export default function Card({ title, children }) {
  return (
    <div className="tfg-card">
      { title  && (<div className="tfg-card__heading">
        { title }
      </div>) } 
      { children && (<div className="tfg-card__body">
        { children }
      </div>) }
    </div>
  )
}