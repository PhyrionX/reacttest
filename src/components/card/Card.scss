.tfg-card {
  background-color: white;
  margin-bottom: 20px;
  box-shadow: 1px 0 20px rgba(0,0,0,.08);
  
  &__heading {
    padding: 10px 20px;
    font-size: 22px;
    font-weight: 700;
  }

  &__body {
    padding: 10px 20px;
  }
}