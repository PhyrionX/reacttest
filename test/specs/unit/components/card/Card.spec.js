import React from 'react';
import { render } from '@testing-library/react'
import Card from '../../../../../src/components/card/card';

describe('<Card />', () => {
  it('should mount empty component', () => {
    const { asFragment } = render(<Card />);
    
    expect(asFragment()).toMatchSnapshot();
  })

  it('should mount component with title', () => {
    const { asFragment } = render(<Card title="Hello"/>);
    
    expect(asFragment()).toMatchSnapshot();
  })

  it('should mount component with title and body', () => {
    const { asFragment } = render(<Card title="Hello">Children</Card>);
    
    expect(asFragment()).toMatchSnapshot();
  })
})