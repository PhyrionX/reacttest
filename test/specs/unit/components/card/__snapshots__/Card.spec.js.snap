// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should mount component with title 1`] = `
<DocumentFragment>
  <div
    class="tfg-card"
  >
    <div
      class="tfg-card__heading"
    >
      Hello
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Card /> should mount component with title and body 1`] = `
<DocumentFragment>
  <div
    class="tfg-card"
  >
    <div
      class="tfg-card__heading"
    >
      Hello
    </div>
    <div
      class="tfg-card__body"
    >
      Children
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Card /> should mount empty component 1`] = `
<DocumentFragment>
  <div
    class="tfg-card"
  />
</DocumentFragment>
`;
